config/src/aws-net-std__gnat.adb

```diff
@@ -218,51 +224,80 @@ package body AWS.Net.Std is
 
       Socket.S := new Socket_Hidden;
 
-      Addr_Info := Get_Inet_Addr (Host, Family, Passive => False);
-      Sock_Addr := Addr_Info.Addr;
-      Sock_Addr.Port := Sockets.Port_Type (Port);
-
-      Sockets.Create_Socket
-        (Socket.S.FD, Addr_Info.Addr.Family, Addr_Info.Mode, Addr_Info.Level);
-      Close_On_Exception := True;
-
-      Set_Non_Blocking_Mode (Socket);
-
       declare
          use GNAT.Sockets;
+         Keep_Excp : Exceptions.Exception_Occurrence;
+         Addresses : constant Address_Info_Array :=
+                       Get_Address_Info
+                         (Host, "", To_GNAT (Family), Passive => False);
       begin
-         Connect_Socket (Socket.S.FD, Sock_Addr);
-      exception
-         when E : GNAT.Sockets.Socket_Error =>
-            --  Ignore EWOULDBLOCK and EINPROGRESS errors, because we are
-            --  using none blocking connect.
-            --  ??? Note, we should change this when GNAT will support
-            --  Non-blocking connect.
-
-            case Resolve_Exception (E) is
-               when Operation_Now_In_Progress
-                    | Resource_Temporarily_Unavailable => null;
-               when others =>
-                  Raise_Error (Exception_Message (E));
-            end case;
-      end;
+         for Addr_Info of Addresses loop
+            begin
```

Comment:
this block is big, maybe we should name it?
Check_Address : begin
 ...
end Check_Address;

Reply:
Done